static/css/components/header-bar.less

```
@@ -441,12 +441,25 @@
 
     .search-bar-submit {
       background: url(/static/images/search-icon.svg) center no-repeat;
-      width: 45px;
+      width: 35px;
       height: 45px;
       border-top-left-radius: .3em;
       border-bottom-left-radius: .3em;
     }
 
+    .vertical-separator {
+      border-left: 1px solid @grey;
+      height: 34px;
+      margin: auto 5px auto 0
+    }
+
+    .search-by-barcode-submit {
+      background: url(/static/images/icons/barcode_scanner.svg) center no-repeat;
+      width: 28px;
+      height: 42px;
+      margin-right: 8px;
+    }
```

Make the icon a little smaller; and use default height so that it's vertically centred.

```suggestion
    .search-by-barcode-submit {
      background: url(/static/images/icons/barcode_scanner.svg) center no-repeat;
      width: 28px;
      background-size: 26px;
      margin-right: 8px;
    }
```